lib/ecto_ldap/adapter.ex

```diff
@@ -37,27 +38,48 @@ defmodule Ecto.Ldap.Adapter do
   # GenServer server API
   #
   ####
-  def handle_call({:search, search_options}, _from, state) do
+  def handle_call({:search, search_options, params}, _from, state) do
+    {:ok, handle}   = ldap_connect(state)
+    search_response = ldap_search(handle, search_options, params)
+    ldap_disconnect(handle)
+
+    {:reply, search_response, state}
+  end
+
+  def handle_call(:base, _from, state) do
+    base = Keyword.get(state, :base) |> to_char_list
+    {:reply, base, state}
+  end
 
+  def ldap_connect(state) do
     user_dn   = Keyword.get(state, :user_dn)  |> to_char_list
     password  = Keyword.get(state, :password) |> to_char_list
-    host      = Keyword.get(state, :host)   |> to_char_list
+    host      = Keyword.get(state, :server)   |> to_char_list
     port      = Keyword.get(state, :port, 636)
     use_ssl   = Keyword.get(state, :ssl, true)
-
+    
     {:ok, handle} = :eldap.open([host], [{:port, port}, {:ssl, use_ssl}])
     :eldap.simple_bind(handle, user_dn, password)
-    search_response = :eldap.search(handle, search_options)
-    :eldap.close(handle)
-
-    {:reply, search_response, state}
+    {:ok, handle}
   end
 
-  def handle_call(:base, _from, state) do
-    base = Keyword.get(state, :base) |> to_char_list
-    {:reply, base, state}
+  def ldap_search(handle, initial_search_options, params) do
+    filter = Keyword.get(initial_search_options, :filter)
+    search_options = case filter do
+      {:and, [equalityMatch: {:AttributeValueAssertion, _dn, {:^, [], [0]}}]} ->
```

Comment:
Yeah I didn't do any of the `get_by` cases at all...  Definitely on the todo list!